openpgp/packet/signature.go

```diff
@@ -491,6 +513,25 @@ func serializeSubpackets(to []byte, subpackets []outputSubpacket, hashed bool) {
 	return
 }
 
+// Version returns the version of the signature. It defaults to version 4.
+func (sig *Signature) Version() int {
+	if sig.version == 5 {
+		return 5
+	}
+	return 4
+}
+
+// MatchIssuerVersion copies the version from the issuer public key
+// (implementations MUST generate version 5 signatures when using a version 5
+// key, and SHOULD generate V4 signatures with version 4 keys). If the key is
+// V5, it upgrades the signature with the necessary fields.
+func (sig *Signature) MatchIssuerVersion(issuer *PublicKey) {
+	sig.version = issuer.Version()
+	if sig.version == 5 {
+		sig.IssuerKeyFingerprint = issuer.Fingerprint
```

Comment:
I would always add the issuer fingerprint, regardless of version; see https://github.com/openpgpjs/openpgpjs/pull/1097. (And do so directly in `Signature.Sign`.)